### PR TITLE
Sync timers between clients and server

### LINKED ISSUE #64: Sync timer between clients and server
Initial proposal is to have server keep the time and clients ask for the time from server.
We already use sockets to get moves, we can send time too.
If sending time very frequently is overkill, we can send it every time there is a move.

@chirag-shinde @sagarbakhtar @codemayank

Please add solution proposals, so that we can evaluate them and start with development.

### PR DESCRIPTION
**Pull request by Prabhakar-Poudel**
closes: #64

### CODE REVIEW

File: client/src/components/chess-game/ChessGame.js [hunk 1]
```diff
@@ -103,28 +112,29 @@ export default class ChessGame extends React.Component {
...
-      this.playerTwoIntervalId = this.startPlayerTwoTimer();
-    } else {
-      clearInterval(this.playerTwoIntervalId);
-      this.playerOneIntervalId = this.startPlayerOneTimer();
+  resetTimers = () => {
+    this.clearTimers();
+    const { currentPlayer, playerColor } = this.state;
+    if (currentPlayer === BLACK_PLAYER) {
+      if (playerColor === BLACK_PLAYER) {
+        this.playerOneIntervalId = this.startPlayerOneTimer();
```
Comment (codemayank) [inline client/src/components/chess-game/ChessGame.js]: Why can't we just fix playerOneTimer to the black player and playerTwoTimer to the white player?

**Review by codemayank — COMMENTED** [review]
The timers start running immediately after the white player makes first move.
White player's timer should start first and only after the black player has made their first move.

File: server/src/sockets/chess.socket.js [hunk 2]
```diff
@@ -19,17 +19,20 @@ const chessSocket = (server) => {
...
 
     socket.on('pieceMoved', (moveObj) => {
       const opponent = assignOpponentSocket(socket.game, socket.id);
-      io.to(opponent.socketId).emit('movePiece', moveObj);
+      const curTime = game.flipTimer();
+      io.to(opponent.socketId).emit('movePiece', { moveObj, time: curTime });
+      socket.emit('movePiece', { moveObj, time: curTime });
     });
 
     socket.on('disconnect', () => {
```
Comment (codemayank) [inline server/src/sockets/chess.socket.js:38]: The timers should be cleared when one of the players disconnects from the game.

File: client/src/components/chess-game/ChessGame.js (hunk 1 above)
Comment (Prabhakar-Poudel) [reply client/src/components/chess-game/ChessGame.js]: In board context playerOne is different for both players. This will require overall change.

**Comment by Prabhakar-Poudel** [comment]
@codemayank 
Plese review.

**Comment by codemayank** [comment]
> The timers start running immediately after the white player makes first move.
> White player's timer should start first and only after the black player has made their first move.

This issue still exists.

**Comment by Prabhakar-Poudel** [comment]
@codemayank 
> The timers start running immediately after the white player makes first move.
White player's timer should start first and only after the black player has made their first move.

Our server is not keeping track of moves. For now it just forwards the moves to clients. As server is the source of truth for time. There is no correlation between number of moves and timers movement. There can only be hacky solution to this, unless server can decide look at moves and decide what to do.

**Review by codemayank — APPROVED** [review]

**Pull request merged**